pkg/admissioncontroller/webhook/admission/authorizationconfig/handler.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	if err = h.getOldObject(request, oldConfigMap); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	if apiequality.Semantic.DeepEqual(configMap.Data, oldConfigMap.Data) {
+		return admissionwebhook.Allowed("authorization configuration or kubeconfig secret names not changed")
+	}
+
+	if errCode, err := validateAuthorizationConfigurationSemantics(authorizationConfiguration, nil, false); err != nil {
+		return admission.Errored(errCode, err)
+	}
+
+	return admissionwebhook.Allowed("ConfigMap change is valid")
+}
+
+func (h *Handler) getOldObject(request admission.Request, oldObj runtime.Object) error {
+	if len(request.OldObject.Raw) != 0 {
+		return h.Decoder.DecodeRaw(request.OldObject, oldObj)
+	}
+	return errors.New("could not find old object")
+}
+
+func validateAuthorizationConfigurationSemantics(authorizationConfiguration string, kubeconfigReferences []gardencore.AuthorizerKubeconfigReference, verifyKubeconfigReferences bool) (errCode int32, err error) {
+	authConfigObj, schemaVersion, err := configDecoder.Decode([]byte(authorizationConfiguration), nil, nil)
+	if err != nil {
+		return http.StatusUnprocessableEntity, fmt.Errorf("failed to decode the provided authorization configuration: %w", err)
+	}
+	authConfig, ok := authConfigObj.(*apiserver.AuthorizationConfiguration)
+	if !ok {
+		return http.StatusInternalServerError, fmt.Errorf("failure to cast to authorization configuration type: %v", schemaVersion)
+	}
+
+	var authorizerNames []string
+
+	for i, webhook := range authConfig.Authorizers {
+		if webhook.Type == apiserver.TypeWebhook && webhook.Webhook != nil {
+			authorizerNames = append(authorizerNames, webhook.Name)
+			// We do not allow users to set the connection info in the webhook configurations, so let's first validate this:
+			if !apiequality.Semantic.DeepEqual(webhook.Webhook.ConnectionInfo, apiserver.WebhookConnectionInfo{}) {
+				return http.StatusUnprocessableEntity, fmt.Errorf("provided invalid authorization configuration: %v", field.ErrorList{field.Forbidden(field.NewPath("authorizers").Index(i).Child("connectionInfo"), "connectionInfo is not allowed to be set")})
+			}
+			// At this point, we have ensured that the connection info is not set. However, below validation function
+			// expects it to be set, so let's fake it here to make the function pass (this is not persisted anywhere).
+			authConfig.Authorizers[i].Webhook.ConnectionInfo.Type = apiserver.AuthorizationWebhookConnectionInfoTypeInCluster
+		}
+	}
+
+	if verifyKubeconfigReferences {
+		for _, authorizerName := range authorizerNames {
+			if !slices.ContainsFunc(kubeconfigReferences, func(ref gardencore.AuthorizerKubeconfigReference) bool {
+				return ref.AuthorizerName == authorizerName
+			}) {
+				return http.StatusUnprocessableEntity, fmt.Errorf("provided invalid authorization configuration: must provide kubeconfig secret name reference for webhook authorizer %q", authorizerName)
```

Comment:
Where is the kubeconfig secret checked, i.e. the secret exists + `data["kubeconfig"]` is available?

Reply:
> Where is the kubeconfig secret checked, i.e. the secret exists 

Good question, I don't think we have such check for several fields, see https://github.com/gardener/gardener/blob/24f0bea3fdfa8def827b882f8a2147ccdb384a07/plugin/pkg/global/resourcereferencemanager/admission.go#L883-L974 I've now added it.

> + `data["kubeconfig"]` is available?

This is covered by the generic https://github.com/gardener/gardener/tree/master/pkg/admissioncontroller/webhook/admission/kubeconfigsecret handler.